controllers/auth-controller.js

```diff
@@ -8,33 +8,49 @@ import gravatar from "gravatar";
 import fs from "fs/promises";
 import path from "path";
 import Jimp from "jimp";
+import { v4 as uuid4 } from "uuid";
 
 const avatarPath = path.resolve("public", "avatars");
 
-const { JWT_SECRET } = process.env;
+const { JWT_SECRET, BASE_URL } = process.env;
 
 const signUp = async (req, res) => {
   const { password, email } = req.body;
   const user = await User.findOne({ email });
 
   if (user) {
-    throw HttpError(409, "Email is already in use");
+    throw HttpError(
+      409,
+      "Email In Use: The provided email address is already associated with an existing account. If you already have an account, please log in."
+    );
   }
   const hashPassword = await bcrypt.hash(password, 10);
   const avatarURL = gravatar.url(email, { s: "200", r: "pg", d: "mm" });
+  const verificationToken = uuid4();
+  console.log("verification code", verificationToken);
   const newUser = await User.create({
     email,
     password: hashPassword,
     avatarURL,
+    verificationToken,
   });
 
+  const verifyEmail = {
+    to: email,
+    subject: "Action Required: Verify Your Email",
+    html: `<a target="_blank" href="${BASE_URL}/api/users/verify/${verificationToken}">Click to verify email</a>`,
+  };
+
+  await sendEmail(verifyEmail);
+
   res.status(201).json({
     message:
-      "Registration successful. You can now sign in with your new account.",
+      "Congratulations! Your registration was successful. You can now sign in with your new account.",
```

Comment:
Here send information that he has to confirm email address.

 Ta osoba nie bedzie się mogła zalogować jeśli tego nie zrobi.